roles/deploy-aws-infra/tasks/main.yml

```diff
@@ -0,0 +1,343 @@
+---
+# tasks file for deploy-infra
+- name: Create VPC if required
+  ec2_vpc_net:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "{{ env_id }}-vpc"
+    cidr_block: 172.31.0.0/16
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+  when: aws_create_vpc
+  register: new_vpc
+
+- name: Create Subnets for the new VPC (HA Mode)
+  ec2_vpc_subnet:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    cidr: 172.31.{{ item }}.0/20
+    region: "{{ aws_region }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    map_public: yes
+    az: "{{ aws_region }}a"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+  with_sequence: start=0 end=64 stride=16
+  when: aws_create_vpc and ha_mode
+  register: new_subnets
+
+- name: Create Subnet for the new VPC (Non HA Mode)
+  ec2_vpc_subnet:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    cidr: 172.31.0.0/20
+    region: "{{ aws_region }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    map_public: yes
+    az: "{{ aws_region }}a"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+  when: aws_create_vpc and not ha_mode
+  register: new_subnets
+
+- name: Create SSH Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ssh"
+    description: "Labs SSH"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+    rules:
+      - proto: tcp
+        from_port: 22
+        to_port: 22
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ssh_sg
+
+- name: Create OCP Master Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ocp-master"
+    description: "Labs OCP Master"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+    rules:
```

Comment:
Can we please source this from a variable and set them in the defaults so we can easily override if needed?

Reply:
What is the advantage of doing that? We can't really re-use this task to remove the instance if exist with "state: absent" as there are some not "compatible" parameters there and also the instances have termination protection so that would fail

Reply:
Not sure if your response was meant for this comment? But basically, I'd like to have the option to specify a different set of security group rules - hence the variable request?

Reply:
@oybed  Sorry, I was just able to view the last 4 lines from the commented code. So, I don't really understand the requirement. Why should we provide the possibility to extend/modify these security groups? As this is an OCP requirement we should ensure that the prereqs are followed, and in case the user needs "extra" SG he/she should be able to add those after the Cluster has been deployed. My main concern here is to avoid OCP installation to fail due to a misconfigured firewall rules.

Reply:
@makentenza ok - not worth spending time on now, we can do a follow-up PR.